modules/Bio/EnsEMBL/Utils/IO/ReportSerializer.pm

```diff
@@ -0,0 +1,221 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+=pod
+
+
+=head1 NAME
+
+Report Serializer - generating textual summary reports
+
+=head1 SYNOPSIS
+
+    use Bio::EnsEMBL::Registry;
+    use Bio::EnsEMBL::Utils::IO::ReportSerializer;
+    use IO::File;
+    
+    my $registry = 'Bio::EnsEMBL::Registry';
+    $output_fh = IO::File->new($config{'output'},'w') or die;
+    $serializer = new ReportSerializer($output_fh);
+    my $slice_adaptor = $registry->get_adaptor( 'Human', 'Core', 'Slice' );
+    my $slice = $slice_adaptor->fetch_by_toplevel_location("6:1000000..1500000");
+    
+    $serializer->print_section_header($slice);
+    $serializer->print_feature_list($slice->get_all_Genes);
+
+=head1 DESCRIPTION
+
+Subclass of Serializer that can turn a feature into a text block
+Unsuited to very large slices, because it requires a select-all approach for features.
+
+=cut
+
+package Bio::EnsEMBL::Utils::IO::ReportSerializer;
+use strict;
+use warnings;
+use Bio::EnsEMBL::Utils::Exception;
+use URI::Escape;
+use Bio::EnsEMBL::Utils::IO::FeatureSerializer;
+
+use base qw(Bio::EnsEMBL::Utils::IO::FeatureSerializer);
+
+my %strand_conversion = ( '1' => '+', '0' => '?', '-1' => '-');
+
+my %feature_conversion = (     'Bio::EnsEMBL::Gene' => 'Gene',
```

Comment:
Values could be obtained with a one-liner, i.e. map combined with split/join

Reply:
For no gain except somebody had to write and test code instead of leaving this as it is.